### PR TITLE
fix(jans-linuux-setup): KC setup issues

### LINKED ISSUE #8210: fix(jans-linux-setup): improper scim configuration for kc
There are various issues preventing SAML to function properly in KC at the moment. 
- the `keycloak.conf` file does not contain the scim configuration data for auth 
- the scim client used by jans has an invalid configuration 
- user federation (storage) isn't properly configured

### LINKED ISSUE #8248: fix(jans-linux-setup):  add more roles to the jans api user created in keycloak and remove duplicate role assignment.
This bug fix is as an addition to issue #7399

We need to increase the roles the user used for API access `jans` has. 
It's been done  [here](https://github.com/JanssenProject/jans/blob/fefbe5f7d74c7b3c0e54c64613cbac2a88cb24c2/jans-linux-setup/jans_setup/setup_app/installers/jans_saml.py#L227) . 
The current roles `manage-identity-providers` and `view-identity-providers`.

TODO LIST : 

- The role `view-identity-providers` is specified twice. Please remove the second reference
- The following additional roles should be assigned to the user: 
  - `query-realms` , `view-realm`
  - `view-clients`, `manage-clients`, `query-clients`, 
  - `query-users` , `view-users`

### PR DESCRIPTION
**Pull request by devrimyatar**
closes #8248
closes #8210

### CODE REVIEW

File: jans-linux-setup/jans_setup/templates/jans-saml/kc_jans_api/jans.userstorage-provider-component.json
Comment (uprightech) [inline jans-linux-setup/jans_setup/templates/jans-saml/kc_jans_api/jans.userstorage-provider-component.json:1]: Not sure if the `jans_idp_realm` refers to the realm name , but it should be the realm's ID

File: jans-linux-setup/jans_setup/templates/jans-saml/keycloak.conf [hunk 1]
```diff
@@ -44,20 +44,20 @@
...
 # scim user search endpoint 
-spi-storage-kc-jans-storage-scim-user-search-endpoint=
+spi-storage-kc-jans-storage-scim-user-search-endpoint=https://%(hostname)s/jans-scim/restv1/v2/Users/.search
 
 # scim oauth scopes
 spi-storage-kc-jans-storage-scim-oauth-scopes=https://jans.io/scim/users.read https://jans.io/scim/users.write
 
 #scim client id 
-spi-storage-kc-jans-storage-scim-client-id=
+spi-storage-kc-jans-storage-scim-client-id=%(scim_client_id)s
```
Comment (uprightech) [inline jans-linux-setup/jans_setup/templates/jans-saml/keycloak.conf]: This should be `saml_scim_client_id` and not `scim_client_id`. We created a SCIM client just for SAML

File: jans-linux-setup/jans_setup/templates/jans-saml/keycloak.conf [hunk 2]
```diff
@@ -44,20 +44,20 @@
...
 
 #scim client secret
-spi-storage-kc-jans-storage-scim-client-secret=
+spi-storage-kc-jans-storage-scim-client-secret=%(scim_client_pw)s
```
Comment (uprightech) [inline jans-linux-setup/jans_setup/templates/jans-saml/keycloak.conf]: This should be `saml_scim_client_pw`. We created aSCIM client just for SAML

File: jans-linux-setup/jans_setup/templates/jans-saml/kc_jans_api/jans.userstorage-provider-component.json
Comment (devrimyatar) [reply jans-linux-setup/jans_setup/templates/jans-saml/kc_jans_api/jans.userstorage-provider-component.json:1]: This is the command executed by setup:
```
/opt/keycloak/bin/kcadm.sh create components -r jans -f /opt/jans/jans-setup/output/jans-saml/kc_jans_api/jans.userstorage-provider-component.json --config /tmp/tmprd4u2bbt/kcadm-jans.config
```

File: jans-linux-setup/jans_setup/templates/jans-saml/keycloak.conf (hunk 2 above)
Comment (devrimyatar) [reply jans-linux-setup/jans_setup/templates/jans-saml/keycloak.conf]: changed

**Review by uprightech — APPROVED** [review]

**Review by yuriyz — APPROVED** [review]

**Pull request merged**